tiatoolbox/tools/registration/wsi_registration.py

```diff
@@ -539,3 +542,314 @@ def estimate_affine_transform(
         matrix[-1, :] = [0, 0, 1]
 
         return matrix
+
+    @staticmethod
+    def get_tissue_regions(
```

Comment:
Can we not do it using WSIReader?

Reply:
I don't think we can use it for two reasons. First, we don't have a function in WSIReader that extracts tissue regions using a binary mask. Second, get_tissue_regions function would be called on a pre-aligned pair of images since pre-alignment is mandatory for DFBR for a successful registration. And, using WSIReader we cannot apply the geometrical transformation.

Reply:
How about PatchExtraction?
https://github.com/TissueImageAnalytics/tiatoolbox/blob/62dc5bdb3e9c9359b667f48da80073285d1772f9/tiatoolbox/tools/patchextraction.py#L117